iothub/device/src/DeviceAuthenticationWithX509Certificate.cs

```diff
@@ -82,14 +82,10 @@ private void SetDeviceId(string deviceId)
         }
 
         /// <summary>
-        /// Dispose the X509 certificate associated with this authentication method.
+        /// The <see cref="Certificate"/> managed resource should be disposed by the user.
         /// </summary>
         public void Dispose()
         {
```

Comment:
This was an oversight, we shouldn't really be disposing the supplied certificates, since the calling application might want to reuse it elsewhere.

Reply:
Without having this, how can a customer do something like `using new DeviceAuthenticationWithX509Certificate()`
The dispose method is not being called from any of our clients from what I know.

Reply:
yes, but what about the scenario where an application reuses the x509 certificate instance for creating multiple objects, one of them being a `DeviceAuthenticationWithX509Certificate` instance. Then, if we dispose the supplied certificate when this auth instance is disposed, the other objects that had a reference to that certificate would be rendered unusable.

Reply:
The way I see it, the user is the owner of the IDisposable type (the x509certificate), so they should decide when it is ready to be disposed.

Reply:
Going by the usage below, it makes sense to not implement the dispose here so that customer can reuse the cert directly in some other place. This also means there is no reason for this class to be IDisposable. However, I guess it is a breaking change so we will just have to stick with a blank implementation. It is worth adding these comments here to the dispose method. At the moment noone is calling this dispose, so it should be ok and unless the customer does using new DeviceAuthenticationWithX509Certificate(), their certificate will not be disposed.

Reply:
sure thing, I'll add a comment explaining why we are not disposing the certificate in here.